sed/calibrator/energy.py

```diff
@@ -1,5 +1,1900 @@
+
+        1. Initialize with all the file paths in a list
+        1a. Use an hdf5 file containing all binned traces and tof
+        1b. Use a mat file containing all binned traces and tof
+        1c. Use the raw data hdf5 files
+        2. Initialize with the folder path containing any of the above files
+        3. Initialize with the binned traces and the time-of-flight
+        """
+
+        if biases is not None:
+            self.biases = biases
+        else:
+            self.biases = np.asarray([])
+        if tof is not None:
+            self.tof = tof
+        else:
+            self.tof = np.asarray([])
+        if traces is not None:
+            self.traces = self.traces_normed = traces
+        else:
+            self.traces = self.traces_normed = np.asarray([])
+        self._config = config
+
+        self.featranges: List[Tuple] = []  # Value ranges for feature detection
+        self.peaks: np.ndarray = np.asarray([])
+        self.calibration: Dict[Any, Any] = {}
+
+        self.tof_column = self._config.get("dataframe", {}).get(
+            "tof_column",
+            "t",
+        )
+        self.energy_column = self._config.get("dataframe", {}).get(
+            "energy_column",
+            "E",
+        )
+        self.x_column = self._config.get("dataframe", {}).get("x_column", "X")
+        self.y_column = self._config.get("dataframe", {}).get("y_column", "Y")
+        self.binwidth: float = self._config.get("dataframe", {}).get(
+            "tof_binwidth",
+            4.125e-12,
+        )
+        self.binning: int = self._config.get("dataframe", {}).get(
+            "tof_binning",
+            1,
+        )
+
+        self.tof_fermi = self._config.get("energy", {}).get(
+            "tof_fermi",
+            132250,
+        )
+        self.x_width = self._config.get("energy", {}).get("x_width", (-20, 20))
+        self.y_width = self._config.get("energy", {}).get("y_width", (-20, 20))
+        self.tof_width = self._config.get("energy", {}).get(
+            "tof_width",
+            (-300, 500),
+        )
+        self.color_clip = self._config.get("energy", {}).get("color_clip", 300)
+
+        self.correction: Dict[Any, Any] = self._config.get("energy", {}).get(
+            "correction",
+            {
+                "correction_type": "Lorentzian",
+                "amplitude": 8,
+                "center": (750, 730),
+                "kwds": {"sigma": 920},
+            },
+        )
+
+    @property
+    def ntraces(self) -> int:
+        """The number of loaded/calculated traces."""
+
+        return len(self.traces)
+
+    @property
+    def nranges(self) -> int:
+        """The number of specified feature ranges."""
+
+        return len(self.featranges)
+
+    @property
+    def dup(self) -> int:
+        """The duplication number."""
+
+        return int(np.round(self.nranges / self.ntraces))
+
+    def load_data(
```

Comment:
It's necessary to have a function to load the data into the class after instantiation, because in the workflow the class is instantiated with the creation of the Processor class, and at that time no data is available yet. However, this function should be reused also by the constructor, as I did in the momentum class

Reply:
I also think it is not good practice to directly modify class attributes, but provide a proper function for this.